### PR TITLE
Handle py 3.6.[01] lack of NoReturn

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------

Use string literal type to be more safe for pythons which do not support it.

Alternate solution to: https://github.com/wandb/client/pull/1820

Testing
-------

circleci

TODO:
- [x] run in python3.6.0 just to be sure (runtime, not mypy)

### CODE REVIEW

**Comment by farizrahman4u** [comment]
@ariG23498 Please test this with python 3.6.0 and 3.6.1. Thanks.

**Comment by ariG23498** [comment]
These were the following steps that I did:
1. using `pyenv` installed python `3.6.0` and `3.6.1`
2. used the `pip install git+https://github.com/wandb/client@fix/py360-noreturn` to install the wandb package from the PR

Both the versions seems to be getting the imports correctly!
image
image

File: wandb/sdk/wandb_run.py [hunk 1]
```diff
@@ -79,6 +78,11 @@
     from .wandb_setup import _WandbSetup
     from wandb.apis.public import Api as PublicApi
 
+    from typing import TYPE_CHECKING
```
Comment (vanpelt) [inline wandb/sdk/wandb_run.py:81]: This looks magical, this is just a constant inside of the `typing` module?

**Review by vanpelt — APPROVED** [review]

File: wandb/sdk/wandb_run.py (hunk 1 above)
Comment (raubitsj) [reply wandb/sdk/wandb_run.py:81]: Yep, magic, and the canonical way this is done with type checkers. 

I will rename wandb.TYPE_CHECKING eventually so it is not as easily confused for typing magic.

**Pull request merged**